.studio/common

```diff
@@ -64,10 +64,27 @@ function compile_go_protobuf_component() {
               return 1
       esac
   fi
+
+  if [ "$1" == "api" ]; then
+    log_line "compiling client mocks"
+    compile_client_mocks
+  fi
 }
 # Adding auto tab complete
 complete -F _component_auto_complete compile_go_protobuf_component
 
+
+function compile_client_mocks() {
+  install_gomock
+  pushd "$scaffolding_go_pkg_path/api/interservice" >/dev/null
+    mockgen -source authz/v2/project.pb.go -destination authz/v2/project.pb.client_mock.go -package v2
+    mockgen -source authz/authz.pb.go -destination authz/authz.pb.client_mock.go -package authz
+    mockgen -source authn/authenticate.pb.go -destination authn/authenticate.pb.client_mock.go -package authn
+    mockgen -source cfgmgmt/service/cfgmgmt.pb.go -destination cfgmgmt/service/cfgmgmt.pb.client_mock.go -package service
+  popd >/dev/null
+}
```

Comment:
@msorens I was trying to follow how the server mocks are currently working. The server mocks are in the same package and folder as the compiled server functions. I do like it better this way because one can easily see the mock structs directly from the original package.